### PR TITLE
Feature/as 3252 add additional question to eligibility checker refused scenario

### PR DESCRIPTION
**Pull request by nssdesikan**
## Ticket Number

https://pins-ds.atlassian.net/browse/AS-3252

## Description of change
Added an additional question page to eligibility checker - 'Refused' scenario

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [x] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/forms-web-app/__tests__/unit/controllers/eligibility/granted-or-refused-permission.test.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      expect(logger.error).toHaveBeenCalledWith(error);
+
+      expect(res.render).toHaveBeenCalledWith(VIEW.ELIGIBILITY.GRANTED_REFUSED_PERMISSION, {
+        appeal: req.session.appeal,
+        errors: {},
+        errorSummary: [{ text: error.toString(), href: '#' }],
+      });
+    });
+
+    it(`'should redirect to '/${VIEW.ELIGIBILITY.DECISION_DATE}' if 'planningPermissionStatus' is 'refused'`, async () => {
```
Comment (shepda) [inline packages/forms-web-app/__tests__/unit/controllers/eligibility/granted-or-refused-permission.test.js:92]: What should happen if the user has selected one of the other 2 options (granted or nodecisionreceived)? 

If it's a different action then perhaps we need tests for that, if it's the same then perhaps we need to add that to teh test description to make it clearer.

File: packages/forms-web-app/src/validators/eligibility/granted-or-refused-permission.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+];
+
+const ruleHouseholderPlanningPermissionStatus = () => 
+  body('granted-or-refused-permission')
+    .notEmpty()
+    .withMessage(
+      'Select if your planning permission was granted or refused, or if you have not received a decision'
+    )
+    .bail()
+    .isIn(validHouseholderPlanningPermissionStatusOptions);
```
[the remark this answers is not in the document]
Comment (nssdesikan) [reply packages/forms-web-app/src/validators/eligibility/granted-or-refused-permission.js:16]: @shepda Sorry I don't get you. Do you think `.isIn(validHouseholderPlanningPermissionStatusOptions);` does not cover what you have raised?

File: packages/forms-web-app/__tests__/unit/controllers/eligibility/granted-or-refused-permission.test.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      });
+    });
+
+    it('should re-render the template with errors if there is any api call error', async () => {
+      const mockRequest = {
+        ...req,
+        body: {},
+      };
+
+      const error = new Error('Cheers');
```
[the remark this answers is not in the document]
Comment (nssdesikan) [reply packages/forms-web-app/__tests__/unit/controllers/eligibility/granted-or-refused-permission.test.js:73]: Next commit will address this

File: packages/forms-web-app/__tests__/unit/controllers/eligibility/granted-or-refused-permission.test.js (hunk 1 above)
Comment (nssdesikan) [reply packages/forms-web-app/__tests__/unit/controllers/eligibility/granted-or-refused-permission.test.js:92]: Those two scenarios will be covered when these cards - [AS-3291](https://pins-ds.atlassian.net/browse/AS-3291) and [AS-3292](https://pins-ds.atlassian.net/browse/AS-3292) - are played

File: packages/forms-web-app/src/validators/eligibility/granted-or-refused-permission.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (shepda) [reply packages/forms-web-app/src/validators/eligibility/granted-or-refused-permission.js:16]: Yes it does, thanks

File: packages/forms-web-app/__tests__/unit/controllers/eligibility/granted-or-refused-permission.test.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (shepda) [reply packages/forms-web-app/__tests__/unit/controllers/eligibility/granted-or-refused-permission.test.js:73]: When you say the next commit will address this, do you mean in this PR or another?

[the remark this answers is not in the document]
Comment (shepda) [reply packages/forms-web-app/__tests__/unit/controllers/eligibility/granted-or-refused-permission.test.js:73]: When you say the next commit will address this, do you mean in this PR or another?

[the remark this answers is not in the document]
Comment (shepda) [reply packages/forms-web-app/__tests__/unit/controllers/eligibility/granted-or-refused-permission.test.js]: Seen the commit, thanks

**Review by shepda — APPROVED** [review]
LGTM

File: packages/forms-web-app/__tests__/unit/controllers/eligibility/granted-or-refused-permission.test.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (nssdesikan) [reply packages/forms-web-app/__tests__/unit/controllers/eligibility/granted-or-refused-permission.test.js]: Yes in this PR. Commit [316a3a8](https://github.com/Planning-Inspectorate/appeal-planning-decision/pull/1707/commits/316a3a830db3c50f9c82c16465b56c7f49d95133) addresses your comments

**Pull request merged**